### PR TITLE
Hybrid conformer export

### PR DESCRIPTION
**Pull request by borisfom**
# What does this PR do ?

Allows export of Hybrid Models with either RNNT decoder (in two parts) or with CTC decoder as single encoder-decoder module. 
Usage: python scripts/export.py  file.nemo file.onnx --config decoder_type=[ctc|rnnt] 

**Collection**: [ASR]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
python scripts/export.py file.nemo file.onnx --config decoder_type=[ctc|rnnt]

# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.


## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by borisfom** [comment]
Includes https://github.com/NVIDIA/NeMo/pull/6978

File: scripts/export.py [hunk 1]
```diff
@@ -130,10 +139,12 @@ def nemo_export(argv):
         in_args["max_dim"] = args.max_dim
         max_dim = args.max_dim
 
-    if args.cache_support and hasattr(model, "encoder") and hasattr(model.encoder, "export_cache_support"):
-        model.encoder.export_cache_support = True
-        logging.info("Caching support is enabled.")
-        model.encoder.setup_streaming_params()
+    if args.cache_support:
+        model.set_export_config({cache_support: True})
```
Comment (titu1994) [inline scripts/export.py:143]: Should be "cache_support": True?

**Review by titu1994 — COMMENTED** [review]
Overall looks great ! Will try it once before approving

**Review by VahidooX — DISMISSED** [review]

File: scripts/export.py (hunk 1 above)
Comment (borisfom) [reply scripts/export.py:143]: That will work either way - it's bool() inside.

**Review by titu1994 — DISMISSED** [review]
Thanks !

**Review by VahidooX — APPROVED** [review]

**Pull request merged**